lib/read-only.js

```diff
@@ -7,23 +8,33 @@ module.exports = function(Model, options) {
 
   // Make sure emailVerified is not set by creation
   Model.stripReadOnlyProperties = function(ctx, modelInstance, next) {
+
     var body = ctx.req.body;
     if (!body) {
       return next();
     }
+
+    var currentUser = ctx.req.currentUser;
+    if (currentUser && options.allowedRoles) {
+      var roleNames = _.map(currentUser.toObject().roles, 'name');
+      var allowed = _.intersection(options.allowedRoles, roleNames);
+      if (allowed.length) return next();
+    }
+
+    var err = new Error('Unable to update: ' + Model.modelName + ' property is read only');
+    err.statusCode = 403;
+
     var properties = (Object.keys(options).length) ? options : null;
-    if (properties) {
-      debug('Creating %s : Read only properties are %j', Model.modelName, properties);
-      Object.keys(properties).forEach(function(key) {
+    if (!properties) return next(err);
```

Comment:
The error seems inappropriate in general if the goal is to address an empty body and subsequent SQL errors when using SQL connectors.